kratos/utilities/discont_utils.h

```diff
@@ -519,144 +521,95 @@ class DiscontinuousShapeFunctionsUtilities
-			std::vector<Matrix>& rGradientsValue, 
-			boost::numeric::ublas::bounded_matrix<double,3*(2-1), (2+1)>& Nenriched,
-			array_1d<double,3>& edge_areas) //, //and information about the interfase:
-    // array_1d<double,(3)>& face_gauss_N, array_1d<double,(3)>& face_gauss_Nenriched, double& face_Area, array_1d<double,(3)>& face_n ,unsigned int& type_of_cut)
+    static int CalculateDiscontinuousShapeFunctions(const bounded_matrix<double,(2+1), 2 >& rPoints,
+                                                    bounded_matrix<double, (2+1), 2 >& rDN_DX,
+                                                    array_1d<double,(2+1)>& rDistances,
+                                                    array_1d<double,(3*(2-1))>& rVolumes,
+                                                    bounded_matrix<double, 3*(2-1), (2+1) >& rGPShapeFunctionValues,
+                                                    array_1d<double,(3*(2-1))>& rPartitionsSign,
+                                                    std::vector<Matrix>& rGradientsValue,
+                                                    bounded_matrix<double,3*(2-1), (2+1)>& rNenriched,
+                                                    array_1d<double,3>& rEdgeAreas)
     {
         KRATOS_TRY
 
-        const double unsigned_distance0=fabs(rDistances(0));
-        const double unsigned_distance1=fabs(rDistances(1));
-        const double unsigned_distance2=fabs(rDistances(2));
-
-        //we begin by finding the largest distance:
-        double longest_distance=fabs(unsigned_distance0);
-        if (unsigned_distance1>longest_distance)
-            longest_distance=unsigned_distance1;
-        if (unsigned_distance2>longest_distance)
-            longest_distance=unsigned_distance2;
-
-        //Now we set a maximum relative distance
-        const double tolerable_distance =longest_distance*0.001;	// (1/1,000,000 seems to have good results)
-        if (unsigned_distance0<tolerable_distance)
-            rDistances[0]=tolerable_distance;
-        if (unsigned_distance1<tolerable_distance)
-            rDistances[1]=tolerable_distance;
-        if (unsigned_distance2<tolerable_distance)
-            rDistances[2]=tolerable_distance;
+        // TRICK TO AVOID INTERSECTIONS TOO CLOSE TO THE NODES
+        const double unsigned_distance_0 = std::abs(rDistances(0));
+        const double unsigned_distance_1 = std::abs(rDistances(1));
+        const double unsigned_distance_2 = std::abs(rDistances(2));
 
-        //END OF TRICK. REMEMBER TO OVERWRITE THE DISTANCE VARIABLE IN THE ELEMENT IN CASE THESE LINES HAVE MODIFIED THEM (distances)
+        // We begin by finding the largest distance:
+        const double longest_distance = std::max(unsigned_distance_0, std::max(unsigned_distance_1, unsigned_distance_2));
 
+        // Set a maximum relative distance
+        const double tolerable_distance = longest_distance*0.001;	// (1/1,000,000 seems to have good results)
```

Comment:
i think that this should not go into this function, but rather on the user side. This is a comment for the future, not for this PR.

Reply:
I completely agree with you.